### PR TITLE
Q032

### LINKED ISSUE #194: Validation (2017): Q032 (Action Taken Date)
**Edit #:** Q032
**Type:** Quality
**Category:** Loan Application Register (only)
**Column:** Action Taken Date

If action taken type = 1, then action taken date should not equal the date application received.

Additional resources:
* [FFIEC's 2016 HMDA File Spec](https://www.ffiec.gov/hmda/pdf/spec2016.pdf)
* [FFIEC's 2016 HMDA Edits](https://www.ffiec.gov/hmda/pdf/edit2016.pdf)
* [CFPB's 2017 HMDA File Spec](http://www.consumerfinance.gov/hmda/static/for-filers/2017/2017-HMDA-File-Specifications.pdf)

### PR DESCRIPTION
**Pull request by nickgrippin**
Closes #194 

Includes modification of test files to pass this quality check.

Depends on #386 to update SingleLarValidationSpec

### CODE REVIEW

File: validation/src/test/scala/hmda/validation/rules/lar/quality/Q032Spec.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  property("All lars with action taken types not equal to 1 must pass") {
+    forAll(larGen) { lar =>
+      whenever(lar.actionTakenType != 1) {
+        lar.mustPass
+      }
+    }
+  }
+
+  property("Lars with action taken = 1 must have different application and action taken dates") {
+    forAll(larGen, dateGen) { (lar, x) =>
```
Comment (kgudel) [inline validation/src/test/scala/hmda/validation/rules/lar/quality/Q032Spec.scala]: It seems like you might not need to use an independent `dateGen`, you could add 1 to `lar.actionTakenDate` or `lar.applicationDate` which I think would be a bit cleaner. This also applies for the following edit as well.

Comment (nickgrippin) [inline validation/src/test/scala/hmda/validation/rules/lar/quality/Q032Spec.scala]: True, I'll try that approach.

**Comment by nickgrippin** [comment]
Travis already passed.  The latest commit didn't have any changes, so it didn't trigger a new instance of Travis.

**Comment by schbetsy** [comment]
LGTM. Feel free to press the merge button once you fix the conflicts and everything is green.

Bummer that you had to make so many changes to make the test files pass. Might be something to discuss at a dev meeting.

**Pull request merged**